File: scala/scalafind/src/test/scala/scalafind/FileResultTest.scala
package scalafind

import org.junit.Assert.assertEquals
import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}

import java.nio.file.Paths


class FileResultTest extends AnyFunSuite with BeforeAndAfterEach with BeforeAndAfterAll {

  test("test_fileresult_abs_path") {
    val filePath = Paths.get("~/src/xfind/scala/scalafind/src/main/scala/scalafind/FileResult.scala")
    val fileResult = new FileResult(filePath, FileType.Code)
    assertEquals(filePath.toString, fileResult.toString)
  }

  test("test_fileresult_rel_path1") {
    val filePath = Paths.get("./FileResult.scala")
    val fileResult = new FileResult(filePath, FileType.Code)
    assertEquals(filePath.toString, fileResult.toString)
  }

  test("test_fileresult_rel_path2") {
    val filePath = Paths.get("../FileResult.scala")
    val fileResult = new FileResult(filePath, FileType.Code)
    assertEquals(filePath.toString, fileResult.toString)
  }
}

File: php/phpfind/tests/FileResultTest.php
<?php

declare(strict_types=1);

use PHPUnit\Framework\TestCase;

use phpfind\FileResult;
use phpfind\FileType;

class FileResultTest extends TestCase
{
    public function test_file_result_abs_path(): void
    {
        $home = getenv('HOME');
        $file_path = "$home/src/xfind/php/phpfind/src/phpfind/FileResult.php";
        $file_type = FileType::Code;
        $file_result = new FileResult($file_path, $file_type, 0, 0);
        $this->assertEquals("$home/src/xfind/php/phpfind/src/phpfind/FileResult.php",
            $file_result->file_path);
    }

    public function test_file_result_rel_path1(): void
    {
        $file_path = './FileResult.php';
        $file_type = FileType::Code;
        $file_result = new FileResult($file_path, $file_type, 0, 0);
        $this->assertEquals('./FileResult.php', $file_result->file_path);
    }

    public function test_file_result_rel_path2(): void
    {
        $file_path = '../FileResult.php';
        $file_type = FileType::Code;
        $file_result = new FileResult($file_path, $file_type, 0, 0);
        $this->assertEquals('../FileResult.php', $file_result->file_path);
    }
}
